dali/test/python/test_operator_rotate.py

```diff
@@ -0,0 +1,196 @@
+      out = np.uint8(np.clip(out, 0, 255))
+    return out
+
+  return warp_fn
+
+class RotatePipeline(Pipeline):
+    def __init__(self, device, batch_size, output_type, input_type, fixed_size=None, num_threads=3, device_id=0, num_gpus=1):
+        super(RotatePipeline, self).__init__(batch_size, num_threads, device_id, seed=7865, exec_async=False, exec_pipelined=False)
+        self.name = device
+        self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+        self.decode = ops.ImageDecoder(device = "cpu", output_type = types.RGB)
+        if input_type != dali.types.UINT8:
+          self.cast = ops.Cast(device = device, dtype = input_type)
+        else:
+          self.cast = None
+
+        self.rotate = ops.Rotate(device = device, size=fixed_size, angle = 30, fill_value = 42, output_dtype = output_type)
+
+    def define_graph(self):
+        self.jpegs, self.labels = self.input(name = "Reader")
+        images = self.decode(self.jpegs)
+        if self.rotate.device == "gpu":
+          images = images.gpu()
+        if self.cast:
+          images = self.cast(images)
+
+        outputs = self.rotate(images)
+        return outputs
+
+class CVPipeline(Pipeline):
+    def __init__(self, batch_size, output_type, input_type, fixed_size, num_threads=3, device_id=0, num_gpus=1):
+        super(CVPipeline, self).__init__(batch_size, num_threads, device_id, seed=7865, exec_async=False, exec_pipelined=False)
+        self.name = "cv"
+        self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+        self.decode = ops.ImageDecoder(device = "cpu", output_type = types.RGB)
+        self.rotate = ops.PythonFunction(function=CVRotate(output_type, input_type, 30, fixed_size))
+        # self.uniform = ops.Uniform(range = (-180.0, 180.0), seed = 42);
+        self.iter = 0
+
+    def define_graph(self):
+        self.jpegs, self.labels = self.input(name = "Reader")
+        images = self.decode(self.jpegs)
+        outputs = self.rotate(images);
+        return outputs
+
+
+def compare(pipe1, pipe2, eps):
+  epoch_size = pipe1.epoch_size("Reader")
+  batch_size = pipe1.batch_size
+  niter = 1 if batch_size >= epoch_size else 2
+  compare_pipelines(pipe1, pipe2, batch_size, niter, eps);
+
+io_types = [
+  (dali.types.UINT8, dali.types.UINT8),
+  (dali.types.UINT8, dali.types.FLOAT),
+  (dali.types.FLOAT, dali.types.UINT8),
+  (dali.types.FLOAT, dali.types.FLOAT)
+]
+
+
+def test_cpu_vs_cv():
+  for batch_size in [1, 4, 19]:
+    for output_size in [None, (160,240)]:
+      for (itype, otype) in io_types:
+        print("Testing cpu vs cv",
+              "\nbatch size: ", batch_size,
+              " output size: ", output_size,
+              " input_type: ", itype,
+              " output_type: ", otype)
+        cv_pipeline = CVPipeline(batch_size, otype, itype, output_size);
+        cv_pipeline.build();
```

Comment:
build() will be invoked in `compare_pipelines` so you don't need to call it here